### PR TITLE
Deployments events handler code refactorization

### PR DESCRIPTION
**Pull request by hks93**
To reduce the number of public functions exported in the index file of deployments-handler API

### CODE REVIEW

File: core/jazz_deployments-event-handler/components/utils.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+/**
+    Helper functions for Events
+    @module: utils.js
+    @author:
+    @version: 1.0
+**/
+const _ = require("lodash");
+const logger = require("../components/logger.js");
+function checkForInterestedEvents(encodedPayload, sequenceNumber, configData) {
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: new line

File: core/jazz_deployments-event-handler/components/utils.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	return new Promise((resolve, reject) => {
+
+    let kinesisPayload = JSON.parse(new Buffer(encodedPayload, 'base64').toString('ascii'));
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: indentation issues here and other places in the code

File: core/jazz_deployments-event-handler/components/utils.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		if (kinesisPayload.Item.EVENT_TYPE && kinesisPayload.Item.EVENT_TYPE.S) {
+			if (_.includes(configData.EVENTS.event_type, kinesisPayload.Item.EVENT_TYPE.S) &&
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: remove this and replace with Array.prototype.includes

File: core/jazz_deployments-event-handler/components/utils.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+					"payload": kinesisPayload.Item
+				});
+			}
+		}
+	});
+};
+
+function getDeploymentPayload(svcContext) {
+	let deploymentPayload = {};
+  svcContext.domain?deploymentPayload.domain = svcContext.domain:null;
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: indentation again in this function

File: core/jazz_deployments-event-handler/components/utils.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+	svcPayload.uri = apiEndpoint;
+	svcPayload.method = method;
+	if (payload) {
+		svcPayload.json = payload;
+	}
+	logger.info("Deployment API payload :" + JSON.stringify(svcPayload));
+	return svcPayload;
+};
+function handleError(errorType, message) {
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: new line please

File: core/jazz_deployments-event-handler/components/utils.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+		},
+		rejectUnauthorized: false,
+		transform: (body, response, resolveWithFullResponse) => {
+			return response;
+		}
+	};
+}
+
+
+const exporatble  = {
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: typo?

File: core/jazz_deployments-event-handler/components/utils.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+// limitations under the License.
+// =========================================================================
+
+/**
+    Helper functions for Events
+    @module: utils.js
+    @author:
+    @version: 1.0
+**/
+const _ = require("lodash");
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: avoid using this

File: core/jazz_deployments-event-handler/components/utils.js (hunk 1 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: done

File: core/jazz_deployments-event-handler/components/utils.js (hunk 5 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: done

File: core/jazz_deployments-event-handler/components/utils.js (hunk 6 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: fixed

File: core/jazz_deployments-event-handler/components/utils.js (hunk 2 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: fixed

File: core/jazz_deployments-event-handler/components/utils.js (hunk 3 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: done

File: core/jazz_deployments-event-handler/components/utils.js [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+          "payload": kinesisPayload.Item
+        });
+      }
+    }
+  });
+};
+
+function getDeploymentPayload(svcContext) {
+  let deploymentPayload = {};
+  svcContext.domain ? deploymentPayload.domain = svcContext.domain : null;
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: better way which is more readable:
const deploymentPayload = {
      domain : svcContext.domain ?  svcContext.domain : null,
    .......
}
what do you think?

File: core/jazz_deployments-event-handler/components/utils.js [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+function getSvcPayload(method, payload, apiEndpoint, authToken) {
+  let svcPayload = {
+    headers: {
+      'content-type': "application/json",
+      'authorization': authToken
+    },
+    rejectUnauthorized: false
+  }
+
+  svcPayload.uri = apiEndpoint;
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: why not make it part of object initializer like in lines 66-70? why doing this separately?

File: core/jazz_deployments-event-handler/components/utils.js [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+  svcPayload.method = method;
+  if (payload) {
+    svcPayload.json = payload;
+  }
+  logger.info("Deployment API payload :" + JSON.stringify(svcPayload));
+  return svcPayload;
+};
+
+function handleError(errorType, message) {
+  let error = {};
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: try to use object initialization
const error = {
failure_code:  errorType,
failure_message: message
}

File: core/jazz_deployments-event-handler/components/utils.js [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+function getTokenRequest(configData) {
+  return {
+    uri: configData.BASE_API_URL + configData.TOKEN_URL,
+    method: 'post',
+    json: {
+      "username": configData.SERVICE_USER,
+      "password": configData.TOKEN_CREDS
+    },
+    rejectUnauthorized: false,
+    transform: (body, response, resolveWithFullResponse) => {
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/components/utils.js]: what purpose is this method solving

File: core/jazz_deployments-event-handler/components/utils.js [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+      if (configData.EVENTS.event_type.includes(kinesisPayload.Item.EVENT_TYPE.S) &&
+          configData.EVENTS.event_name.includes(kinesisPayload.Item.EVENT_NAME.S)) {
+        logger.info("found " + kinesisPayload.Item.EVENT_TYPE.S + " event with sequence number: " + sequenceNumber);
+
+        return resolve({
+          "interested_event": true,
+          "payload": kinesisPayload.Item
+        });
+      } else {
+        logger.error("Not an interested event or event type");
```
Comment (suryajak) [inline core/jazz_deployments-event-handler/components/utils.js]: I know this is a carry over from other handlers, but make this `debug`

File: core/jazz_deployments-event-handler/components/utils.js [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+    },
+    rejectUnauthorized: false
+  }
+
+  svcPayload.uri = apiEndpoint;
+  svcPayload.method = method;
+  if (payload) {
+    svcPayload.json = payload;
+  }
+  logger.info("Deployment API payload :" + JSON.stringify(svcPayload));
```
Comment (suryajak) [inline core/jazz_deployments-event-handler/components/utils.js]: `logger.debug`

File: core/jazz_deployments-event-handler/components/utils.js (hunk 10 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: done

File: core/jazz_deployments-event-handler/components/utils.js (hunk 12 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: done

File: core/jazz_deployments-event-handler/components/utils.js (hunk 9 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: fixed

File: core/jazz_deployments-event-handler/components/utils.js (hunk 13 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: done

File: core/jazz_deployments-event-handler/components/utils.js (hunk 11 above)
Comment (hks93) [reply core/jazz_deployments-event-handler/components/utils.js]: it is creating the payload for the request-promise call

File: core/jazz_deployments-event-handler/components/utils.js (hunk 8 above)
Comment (svidhani) [reply core/jazz_deployments-event-handler/components/utils.js]: you are right

File: core/jazz_deployments-event-handler/components/utils.js (hunk 11 above)
Comment (svidhani) [reply core/jazz_deployments-event-handler/components/utils.js]: Well the reason i was asking is its just returning response and not doing much. It says 'transform' but there is no transformation

File: core/jazz_deployments-event-handler/index.js [hunk 14]
```diff
@@ -374,22 +270,29 @@ function handleFailedEvents(id, failure_message, payload, failure_code) {
 		"failure_code": failure_code,
 		"failure_message": failure_message
 	});
+}
+function processRequest(svcPayload) {
+	return new Promise((resolve, reject) => {
+		request(svcPayload, function (error, response, body) {
```
Comment (svidhani) [inline core/jazz_deployments-event-handler/index.js]: i see the indentation here is diff than in other files.

**Pull request merged**